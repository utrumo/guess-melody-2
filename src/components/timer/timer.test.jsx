import React from 'react';
import renderer from 'react-test-renderer';

import Timer from './timer.jsx';

it(`Timer correctly renders`, () => {
  const tree = renderer.create(<Timer time={30000} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it(`Timer correctly renders if time < 30 sec`, () => {
  const tree = renderer.create(<Timer time={29000} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it(`Timer correctly renders if minutes and seconds < 10`, () => {
  const time = 9 * 60 * 1000 + 9000;
  const tree = renderer.create(<Timer time={time} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it(`Timer correctly renders if time over`, () => {
  const tree = renderer.create(<Timer time={0} />).toJSON();
  expect(tree).toMatchSnapshot();
});
