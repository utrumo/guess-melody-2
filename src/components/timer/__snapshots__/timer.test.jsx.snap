// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer correctly renders 1`] = `
<div
  className="timer__value"
  xmlns="http://www.w3.org/1999/xhtml"
>
  <span
    className="timer__mins"
  >
    00
  </span>
  <span
    className="timer__dots"
  >
    :
  </span>
  <span
    className="timer__secs"
  >
    30
  </span>
</div>
`;

exports[`Timer correctly renders if minutes and seconds < 10 1`] = `
<div
  className="timer__value"
  xmlns="http://www.w3.org/1999/xhtml"
>
  <span
    className="timer__mins"
  >
    09
  </span>
  <span
    className="timer__dots"
  >
    :
  </span>
  <span
    className="timer__secs"
  >
    09
  </span>
</div>
`;

exports[`Timer correctly renders if time < 30 sec 1`] = `
<div
  className="timer__value timer__value--warning"
  xmlns="http://www.w3.org/1999/xhtml"
>
  <span
    className="timer__mins"
  >
    00
  </span>
  <span
    className="timer__dots"
  >
    :
  </span>
  <span
    className="timer__secs"
  >
    29
  </span>
</div>
`;

exports[`Timer correctly renders if time over 1`] = `
<div
  className="timer__value timer__value--finished"
  xmlns="http://www.w3.org/1999/xhtml"
>
  <span
    className="timer__mins"
  >
    00
  </span>
  <span
    className="timer__dots"
  >
    :
  </span>
  <span
    className="timer__secs"
  >
    00
  </span>
</div>
`;
